### ISSUE TITLE
Users and Groups HA/CARP not adding users

### ISSUE THREAD

**Issue by CoMPaTech** [opnsense/core#2151]
Hi OPNsense,

Adding 'users & groups' to the High Availability https://.../system_hasync.php does replicate the users data (OTP, access keys, etc.) to the other node, but no user is created in the operating system.

OPNsense 17.7.12_1

Reproduced on testing system by enabling HA/CARP according to the OPNsense documentation and then adding 'user and groups' to the list of synced items. Users are replicated, but only in the WUI not in /etc/passwd etc.

**Comment by fichtner** [comment]
Hi @CoMPaTech,

I don't use HA, but I think this may be a support case: configurations are synced, but the filter / firewall needs to be reloaded from the other end manually in this case, otherwise the config is not set as active right away.


Cheers,
Franco

**Comment by CoMPaTech** [comment]
In our humble opinion users should be created on both sides, but that might be a difference of opinion. All other HA aspects seem to work fine though, just users aren't synced, but we could try and failover and see if users are created (on OS level) - will get back on that.

**Comment by fichtner** [comment]
Not trying to argue, only stating what I know. @AdSchellevis knows more.

**Comment by CoMPaTech** [comment]
Sorry, wasn't trying to create an arguement ... apologies!

On the issue at hand, we just tried adding another user and is has to do with rebooting. If we create the user on the (active/designated) master (fw03) and then **reboot** the backup (fw04) then the users are created for CLI access (/etc/passwd) just fine.

Expected behaviour would have been that the reboot would not be required for this.

**Comment by AdSchellevis** [comment]
We should probably add a "service" which you can restart using the status page, just like the other real services. The synchronisation script only diffs the two configs and writes the changes, for filter reload there's a hook in https://github.com/opnsense/core/blob/18.1.2/src/etc/rc.filter_synchronize to trigger the filter configure.

The actions associated to the config can't be triggered automatically, because it's virtually impossible to determine the correct action from the xml diff, hence all the other reloads should be triggered by the user after he's done configuring the master firewall. 

Long story short, we can make sure you don't need a reboot, we can't make sure it's handled automatically without getting into a maintenance issue.

**Comment by CoMPaTech** [comment]
Manual service reload should be just fine - there are other actions that also 'pop-up/modal' the need for manual action if I recall correctly.
Rebooting a designated backup isn't that big a deal and acceptable as a workaround while the 'service' is in-progress :)

**Comment by AdSchellevis** [comment]
ok, let's put that on our list to do then.

**Comment by AdSchellevis** [comment]
@CoMPaTech https://github.com/opnsense/core/commit/9ca61d8b3e9c8d615c34902a0265d5859bf2126f should add a login "service" which reloads users and groups.

In case you want to test, execute the following on the backup:

`opnsense-patch 9ca61d8
`
